### PR TITLE
_WD_LoadWait - better error handling and logging

### PR DESCRIPTION
**Pull request by mlipok**
### Proposed changes

`_WD_LoadWait` should have better error handling and logging

### Checklist

- [x] I have read and noticed the [CODE OF CONDUCT](https://github.com/Danp2/au3WebDriver/blob/master/docs/CODE_OF_CONDUCT.md) document
- [x] I have read and noticed the [CONTRIBUTING](https://github.com/Danp2/au3WebDriver/blob/master/docs/CONTRIBUTING.md) document
- [ ] I have added necessary documentation or screenshots (if appropriate)

### Types of changes

Please check `x` the type of change your PR introduces:

- [ ] Bugfix (change which fixes an issue)
- [x] Feature (change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (functional, structural)
- [x] Documentation content changes
- [ ] Other (please describe)

### What is the current behavior?
On `$_WD_DEBUG_Error` function do not set complete [information about `$iTimeout` reason](https://www.w3schools.com/jsref/prop_doc_readystate.asp).

`_WD_ExecuteScript($sSession, 'return document.readyState', '', Default, $_WD_JSON_Value)` can set error to <> `$_WD_ERROR_Timeout` what was not documented in header.

### What is the new behavior?

added support for `$_WD_ERROR_Exception`

### Additional context

https://www.w3schools.com/jsref/prop_doc_readystate.asp

and `$iExt` shows latest status from: https://www.w3schools.com/jsref/prop_doc_readystate.asp

I was working with https://github.com/Danp2/au3WebDriver/pull/362 and because of this https://github.com/Danp2/au3WebDriver/pull/362#issuecomment-1225545526 I was thinking about small refactoring 
even before we decide about `it would be helpfull to change _WD_LoadWait behavior on the fly without touching capabilities.` : [_WD_LoadWait() enhancements](https://github.com/Danp2/au3WebDriver/issues/385)

### System under test

Please complete the following information.

### CODE REVIEW

**Comment by mlipok** [comment]
I'm with your change

**Comment by Danp2** [comment]
Should these string literals (loading, loaded, etc) be changed to constants?

https://github.com/Danp2/au3WebDriver/blob/3b85f07a965fe43902b3175007c8badaf9b6d491/wd_helper.au3#L781-L793

**Comment by mlipok** [comment]
> Should these string literals (loading, loaded, etc) be changed to constants?
> 
> https://github.com/Danp2/au3WebDriver/blob/3b85f07a965fe43902b3175007c8badaf9b6d491/wd_helper.au3#L781-L793

Yes. Even more it can be changed to `Global Const $_WD_ReadyState[]`
What you think about ?

**Comment by Danp2** [comment]
> Global Const $_WD_ReadyState[]

I was thinking something similar, but not an array. How will that work?

**Comment by mlipok** [comment]
> [@mlipok ](/mlipok)
> ` [$aWD_READYSTATE added](/Danp2/au3WebDriver/pull/384/commits/6193d341784b5464e91bdbb4e5901648f6f34bb7)`

Please take a look on: [$aWD_READYSTATE added](https://github.com/Danp2/au3WebDriver/pull/384/commits/6193d341784b5464e91bdbb4e5901648f6f34bb7)

File: wd_helper.au3 [hunk 1]
```diff
@@ -961,7 +980,20 @@ Func _WD_LoadWait($sSession, $iDelay = Default, $iTimeout = Default, $sElement =
...
 			$sReadyState = _WD_ExecuteScript($sSession, 'return document.readyState', '', Default, $_WD_JSON_Value)
 			$iErr = @error
-			If $iErr Or $sReadyState = 'complete' Then
+			If $iErr Then
+				$iErr = $_WD_ERROR_Exception
+				$sReadyState = ''
+			Else
+				For $i = 0 To $_WD_READYSTATE__COUNTER - 1
+					If $sReadyState = $aWD_READYSTATE[$i][0] Then ; https://www.w3schools.com/jsref/prop_doc_readystate.asp
+						$iExt = $i
```
Comment (Danp2) [inline wd_helper.au3:989]: This link doesn't belong here IMO.

Comment (mlipok) [reply wd_helper.au3]: Fixed: [w3schools link](https://github.com/Danp2/au3WebDriver/pull/384/commits/9b5b2b5bb53355fb15695741275297bf6d2b311f)

File: wd_helper.au3 [hunk 2]
```diff
@@ -961,7 +980,20 @@ Func _WD_LoadWait($sSession, $iDelay = Default, $iTimeout = Default, $sElement =
...
+				$sReadyState = ''
+			Else
+				For $i = 0 To $_WD_READYSTATE__COUNTER - 1
+					If $sReadyState = $aWD_READYSTATE[$i][0] Then
+						$iExt = $i
+						$sReadyState &= ' (' & $aWD_READYSTATE[$i][1] & ')'
+						ExitLoop
+					EndIf
+				Next
+			EndIf
```
Comment (Danp2) [inline wd_helper.au3:994]: Did you consider using `_ArraySearch` here?

Comment (mlipok) [reply wd_helper.au3:994]: Please make do the trick, as I go sleep to bed in a moment.

Comment (Danp2) [reply wd_helper.au3]: @mlipok Please take a look at https://github.com/Danp2/au3WebDriver/pull/384/commits/9a969783f9e116f7070f4dc929fb3c80b74568cb and provide your feedback.

File: wd_helper.au3 [hunk 3]
```diff
@@ -983,17 +988,10 @@ Func _WD_LoadWait($sSession, $iDelay = Default, $iTimeout = Default, $sElement =
...
 				$sReadyState = ''
-			Else
-				For $i = 0 To $_WD_READYSTATE__COUNTER - 1
-					If $sReadyState = $aWD_READYSTATE[$i][0] Then
-						$iExt = $i
-						$sReadyState &= ' (' & $aWD_READYSTATE[$i][1] & ')'
-						ExitLoop
-					EndIf
-				Next
+				ExitLoop
```
Comment (mlipok) [inline wd_helper.au3:991]: Why exitloop here?

Comment (mlipok) [reply wd_helper.au3:991]: Never mind I seeniw

File: wd_helper.au3 [hunk 4]
```diff
@@ -961,7 +985,13 @@ Func _WD_LoadWait($sSession, $iDelay = Default, $iTimeout = Default, $sElement =
...
 			$sReadyState = _WD_ExecuteScript($sSession, 'return document.readyState', '', Default, $_WD_JSON_Value)
 			$iErr = @error
-			If $iErr Or $sReadyState = 'complete' Then
+			If $iErr Then
+				$iErr = $_WD_ERROR_Exception
+				$sReadyState = ''
+				ExitLoop
+			EndIf
+
+			If $sReadyState = 'complete' Then
```
Comment (mlipok) [inline wd_helper.au3:994]: Elseif ?

Comment (Danp2) [reply wd_helper.au3:994]: I considered it, but decided not to make the change. I believe the code with work functionally the same either way.

Comment (mlipok) [reply wd_helper.au3:994]: J

File: wd_helper.au3 (hunk 2 above)
Comment (mlipok) [reply wd_helper.au3]: Is ok

File: wd_helper.au3 [hunk 5]
```diff
@@ -976,8 +1006,19 @@ Func _WD_LoadWait($sSession, $iDelay = Default, $iTimeout = Default, $sElement =
 	WEnd
 	$_WD_DEBUG = $_WD_DEBUG_Saved ; restore DEBUG level
 
+	If $sReadyState Then
+		$iIndex = _ArraySearch($aWD_READYSTATE, $sReadyState, Default, Default, Default, Default, Default, $_WD_READYSTATE_State)
+		If @error Then
+			$iErr = $_WD_ERROR_GeneralError
```
Comment (mlipok) [inline wd_helper.au3:1012]: Here I would like to propose `$_WD_ERROR_RetValue` instead

**Pull request merged**